idl/response/foxsi_get_output_image_cube.pro

```diff
@@ -0,0 +1,403 @@
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;FUNCTION:      "foxsi_get_output_image_cube"
+;;;
+;;;HISTORY:       Initial Commit - 08/25/15 - Samuel Badman
+;;;               Improved Convolution Method - 08/31/15 - Samuel Badman               
+;;;
+;;;DESCRIPTION:   Function which takes a simulated event (an array of
+;;;               flux maps at different energy values) and obtains a
+;;;               simulation of the image FOXSI would observe from
+;;;               this event.
+;;;               Each flux map is converted to counts through
+;;;               multiplication by an effective area obtained from
+;;;               foxsi_get_eff_area.pro as a function of the energy
+;;;               of the midpoint of flux maps energy range. 
+;;;               These count maps are then convolved with a point
+;;;               spread function obtained through calling the
+;;;               function get_psf_array with arguments specifying the
+;;;               solar coordinates of the image sensor. After
+;;;               convolution, the new image is rebinned according to
+;;;               the keyword px = pix_size (default = 3) to reflect
+;;;               the loss of resolution due to the finite strip size
+;;;               in the detectors.The output is a structure with
+;;;               identical parameters to the input source with the
+;;;               above appended tags. This contains a processed
+;;;               (convolved and pixelised) image at each energy value
+;;;               of the source along with the relevant spectral information.
+;;; 
+;;;
+;;;CALL SEQUENCE: output_map_cube  = foxsi_get_output_image_cube()
+;;;
+;;;
+;;;KEYWORDS:      source_map_spectrum  = "source_map_spectrum". User
+;;;               inputted source image array, if blank, default 
+;;;               generated from get_source_map_spectrum.pro
+;;;               function
+;;;               
+;;;               bin_min = 'The value of the lower bound of
+;;;                         the lowest energy bin in the user- 
+;;;                         provided spectrum in keV'
+;;;
+;;;               bin_max = 'The value of the upper bound of
+;;;                         the lowest energy bin in the user- 
+;;;                         provided spectrum in keV'
+;;;               
```

Comment:
I recommend e_min and e_max rather than bin_min and bin_max, to explicitly make it clear that it's the lower and upper edges of the lowest and upper bins, rather than, say the midpoints of those bins.  (It's made clear in this preamble, but users may forget.)